lib/Driver/ToolChains.h

```diff
@@ -111,6 +111,16 @@ class LLVM_LIBRARY_VISIBILITY Android : public GenericUnix {
   ~Android() = default;
 };
 
+class LLVM_LIBRARY_VISIBILITY Fuchsia : public GenericUnix {
```

Comment:
This is what we originally did in Clang as well, but it was causing more issues than solving since Fuchsia really isn't UNIX, so we since moved to inheriting directly from the `ToolChain` class, that's something to consider.

Reply:
Yeah, right now there isn't too much friction pretending Fuchsia is a type of "Unix" for at least the args used by Swift to drive clang/ld (Fuchsia is elf platform but not a Unix, even though it has some posix-y looking things). 

It might make sense to make a base that both GenericUnix and Fuchsia's toolchain class inherit called ElfPlatform or something maybe and move things around. 

We have a similar friction with the name of the Glibc module (we aren't using "GNU Glibc" but it's the name that existing code assumes the libc module is called outside of Darwin). I'm hoping people will use our platform libc module map instead there even though the Glibc module works right now too.